ert/storage/_storage.py

```diff
@@ -316,7 +316,7 @@ async def transmit_record_collection(
     workspace_name: str,
     experiment_name: Optional[str] = None,
 ) -> Dict[int, Dict[str, ert.data.RecordTransmitter]]:
-    assert record_coll.ensemble_size is not None
+    assert len(record_coll) is not None
```

Comment:
Releates to the dicussion point I made in `_record.py`, `len` should probably not return `None`.

Reply:
This assert should go, since the class guarantees this is not the case.